docs/sql-programming-guide.md

```diff
@@ -1002,6 +1002,15 @@ Configuration of Parquet can be done using the `setConf` method on `SparkSession
     </p>
   </td>
 </tr>
+<tr>
+  <td><code>spark.sql.parquet.writeLegacyFormat</code></td>
```

Comment:
This should go with the other parquet properties if anything, but, this one is so old I don't think it's worth documenting. It shouldn't be used today.

Reply:
@srowen, actually, this configuration specifically related with compatibility with other systems like Impala (not only old Spark ones) where decimals are written based on fixed binary format (nowdays it's written in int-based in Spark). If this configurations is not enabled, they are unable to read what Spark wrote.

Given https://stackoverflow.com/questions/44279870/why-cant-impala-read-parquet-files-after-spark-sqls-write and JIRA like [SPARK-20297](https://issues.apache.org/jira/browse/SPARK-20297), I think this configuration is kind of important. I even expected more documentation about this configuration specifically at the first place.

Personally I have been thinking it would better to leave this configuration after 3.0 as well for better compatibility.

Reply:
I'd like to add my 2 cents. We use both Spark and Hive in our Hadoop/Spark clusters. And we have 2 types of tables, working tables and target tables. Working tables are only used by Spark jobs, while target tables are populated by Spark and exposed to downstream jobs including Hive jobs. Our data engineers frequently meet with this issue when they use Hive to read target tables. Finally we decided to set spark.sql.parquet.writeLegacyFormat=true as the default value for target tables and explicitly describe this in our internal developer guide.

Reply:
OK that sounds important to document. But the reasoning in this thread is also more useful information I think. Instead of describing it as a legacy format (implying it's not valid Parquet or something) and that it's required for Hive and Impala, can we mention or point to the specific reason that would cause you to need this? The value of the documentation here is in whether it helps the user know when to set it one way or the other.

Reply:
++1 for more information actually.

Reply:
OK, I will update the doc and describe scenarios and reasons why we need this flag.